warp10/src/main/java/io/warp10/script/ext/http/HttpWarpScriptExtension.java

```diff
@@ -0,0 +1,172 @@
+   */
+  public static final String ATTRIBUTE_HTTP_LIMIT = "http.limit";
+  public static final String ATTRIBUTE_HTTP_LIMIT_HARD = "http.limit.hard";
+
+  /**
+   * Number of calls to HTTP so far in the sessions
+   */
+  public static final String ATTRIBUTE_HTTP_COUNT = "http.count";
+
+  /**
+   * Maximum cumulative size of content retrieved via calls to HTTP in a session
+   */
+  public static final String ATTRIBUTE_HTTP_MAXSIZE = "http.maxsize";
+  public static final String ATTRIBUTE_HTTP_MAXSIZE_HARD = "http.maxsize.hard";
+
+  /**
+   * Current  HTTP so far in the sessions
+   */
+  public static final String ATTRIBUTE_HTTP_SIZE = "http.size";
+
+  //
+  // DEFAULTS
+  //
+
+  public static final long DEFAULT_HTTP_LIMIT = 64L;
+  public static final long DEFAULT_HTTP_MAXSIZE = 1000000L;
+
+  //
+  // ASSOCIATIONS attributes to either configuration or defaults
+  //
+
+  /**
+   * Associates the attribute name to the configuration name
+   */
+  private static final Map<String, String> attributeToConf;
+
+  /**
+   * Associates the attribute name to its default value
+   */
+  private static final Map<String, Long> attributeToDefault;
+
+  private static final Map<String, Object> functions;
+
+  static {
+    // Initialize attribute->configuration
+    Map<String, String> a2c = new HashMap<String, String>();
+    a2c.put(ATTRIBUTE_HTTP_LIMIT, WARPSCRIPT_HTTP_LIMIT);
+    a2c.put(ATTRIBUTE_HTTP_MAXSIZE, WARPSCRIPT_HTTP_MAXSIZE);
+    a2c.put(ATTRIBUTE_HTTP_LIMIT_HARD, WARPSCRIPT_HTTP_LIMIT_HARD);
+    a2c.put(ATTRIBUTE_HTTP_MAXSIZE_HARD, WARPSCRIPT_HTTP_MAXSIZE_HARD);
+    attributeToConf = Collections.unmodifiableMap(a2c);
+
+    // Initialize attribute->default
+    Map<String, Long> a2d = new HashMap<String, Long>();
+    a2d.put(ATTRIBUTE_HTTP_LIMIT, DEFAULT_HTTP_LIMIT);
+    a2d.put(ATTRIBUTE_HTTP_MAXSIZE, DEFAULT_HTTP_MAXSIZE);
+    a2d.put(ATTRIBUTE_HTTP_LIMIT_HARD, DEFAULT_HTTP_LIMIT);
+    a2d.put(ATTRIBUTE_HTTP_MAXSIZE_HARD, DEFAULT_HTTP_MAXSIZE);
+    attributeToDefault = Collections.unmodifiableMap(a2d);
+
+    // Create functions and map
+    functions = new HashMap<String, Object>();
+
+    functions.put("HTTP", new HTTP("HTTP"));
+    functions.put("MAXURLCOUNT", new MAXURLCOUNT("MAXURLCOUNT"));
+    functions.put("MAXDOWNLOADSIZE", new MAXDOWNLOADSIZE("MAXDOWNLOADSIZE"));
+  }
+
+  public Map<String, Object> getFunctions() {
+    return functions;
+  }
+
+  /**
+   * Get the value of the attribute in the stack, if not present get the value in the configuration and it not present either, get the default value.
+   * Also set the attribute in the stack once the value is found.
+   *
+   * @param stack     The stack the get the attribute from, if present.
+   * @param attribute The attribute name to get.
+   * @return The first available value in the list: attribute value, configuration value, default.
+   */
+  public static Long getLongAttribute(WarpScriptStack stack, String attribute) {
```

Comment:
Remove this. We can consider adding a method to StackUtils to enforce a similar logic but with a simpler behavior, i.e. no attributeToConf structure but simply an extraneous parameter to the call.